xgboost_ray/main.py

```diff
@@ -855,7 +922,7 @@ def train(params: Dict,
                     f"Sleeping for 10 seconds for cleanup.")
                 start_again = True
 
-            if tries + 1 <= max_actor_restarts:
+            elif tries + 1 <= max_actor_restarts:
```

Comment:
I believe this should be an elif @krfricke? `max_actor_restarts` should be ignored if elastic training is True if I'm not mistaken.

Reply:
Hmm, usually we should still have a max_actor_restart for elastic training. This usually prevents infinite loops for some obscure error.

Reply:
I think there's 2 points here: whether `max_actor_restarts` should have an impact with elastic training and whether line 925 should be an if or elif.

For the first point, no actors are restarted with elastic training, so what would `max_actor_restarts` be used for? Also we say in the docstring that `max_actor_restarts` is ignored during elastic training (which intuitively is what I would expect as a user). But perhaps some other behavior makes more sense here.

But regardless of the first point, I think line 925 should be an elif. Functionally the only change it makes is not logging an incorrect logger.warning message.

Reply:
Thanks for clearing that up, I missed that context. The elif looks right then.

Reply:
It should be an `elif`. I added that somewhere, but it might have been reverted in some merge action before